pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/TriggerSomaticResultSurveyRoute.java

```diff
@@ -0,0 +1,149 @@
+import org.broadinstitute.dsm.service.SomaticResultUploadService;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.DDPRequestUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class TriggerSomaticResultSurveyRoute extends RequestHandler {
+    private final SomaticResultUploadService service;
+    private static final Logger logger = LoggerFactory.getLogger(TriggerSomaticResultSurveyRoute.class);
+    private static final String DEFAULT_TRIGGER_COMMENT = "Somatic results shared with participant.";
+    protected static final String REQUIRED_SURVEY_CREATION_ROLE = "survey_creation";
+    protected static final String REQUIRED_UPLOAD_ROR_ROLE = "upload_ror_file";
+
+    public TriggerSomaticResultSurveyRoute(SomaticResultUploadService service) {
+        this.service = service;
+    }
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = queryParams.get(RoutePath.REALM).value();
+        if (StringUtils.isBlank(realm)) {
+            throw new DSMBadRequestException("realm is a required query parameter.");
+        }
+        if (!isAuthorized(userId, realm)) {
+            throw new AuthorizationException(UserErrorMessages.NO_RIGHTS);
+        }
+        SomaticResultTriggerRequestPayload requestPayload = getRequestPayload(queryParams, request);
+        SomaticResultUpload resultUpload = service.getSomaticResultByIdPtptAndRealm(requestPayload.getSomaticDocumentId(),
+                requestPayload.getParticipantId(), realm);
+        if (resultUpload != null && isEligibleToBeSent(resultUpload)) {
+            DDPInstance instance = DDPInstance.getDDPInstance(realm);
+            long triggerId = buildTriggerComment(userId);
+            SomaticResultUpload triggerUpdatedSomaticUpload =
+                    service.updateSomaticResultTrigger(resultUpload.getSomaticDocumentId(), triggerId, realm);
+            SomaticResultTriggerActivityPayload payloadToSend = new SomaticResultTriggerActivityPayload(
+                    requestPayload.getParticipantId(),
+                    triggerId,
+                    triggerUpdatedSomaticUpload.getBucket(),
+                    triggerUpdatedSomaticUpload.getBlobPath());
+            return DDPRequestUtil.triggerFollowupSurvey(instance, payloadToSend, requestPayload.getSurveyName());
+        } else {
+            throw new DSMBadRequestException("Bad somatic document id.  Not triggering followup");
+        }
+    }
+
+    private boolean isEligibleToBeSent(SomaticResultUpload resultUpload) {
+        if (Boolean.FALSE.equals(resultUpload.getIsVirusFree())) {
```

Comment:
New check to make sure we're only sending results if they're virus free and they haven't been deleted.